server.js

```diff
@@ -20,7 +37,44 @@ app.use(bodyParser.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Happy Thoughts API')
+})
+
+app.get('/thoughts', async (req, res) => {
+  const thoughts = await Thought.find()
+    .sort({createdAt: 'desc'})
+    .limit(20)
+    .exec()
+  res.json(thoughts)
+})
+
+app.post('/thoughts', async (req,res) => {
+  const newThought = new Thought({message: req.body.message})
+  try {
+    const savedThought = await newThought.save()
+    res.status(200).json(savedThought)
+  } catch (err) {
+    res.status(400).json({
+      message : 'Bad Request - Could not save message', 
+      error: err.errors
+      })
+  }
+})
+
+app.post('/thoughts/:id/like', async (req, res) => {
+  try {
+    const updatedThought = await Thought.updateOne(
+      {_id: req.params.id}, 
+      {$inc: {hearts: 1}},
+      {new: true}
+    )
+    res.status(200).json(updatedThought)
+  } catch (err) {
+    res.status(404).json({ 
+      message: 'Bad Request - Could not post like', 
+      error: err.errors
+    })
+  }
 })
 
```

Comment:
Great job Karan with the harts 💯

Reply:
Hi Sandra, Thank you for the review. I have made the requested changes and the problem of 5 characters was in the front end where it was 6 :) Now that is fixed too.

Cheers /K